src/transformers/models/x_clip/processing_x_clip.py

```diff
@@ -99,19 +112,25 @@ def __call__(self, text=None, videos=None, return_tensors=None, **kwargs):
         if text is None and videos is None:
             raise ValueError("You have to specify either text or videos. Both cannot be none.")
 
+        output_kwargs = self._merge_kwargs(
+            XCLIPProcessorKwargs,
+            tokenizer_init_kwargs=self.tokenizer.init_kwargs,
+            **kwargs,
+        )
+
         if text is not None:
-            encoding = self.tokenizer(text, return_tensors=return_tensors, **kwargs)
+            encoding = self.tokenizer(text, **output_kwargs["text_kwargs"])
 
         if videos is not None:
-            image_features = self.image_processor(videos, return_tensors=return_tensors, **kwargs)
+            image_features = self.image_processor(videos, **output_kwargs["videos_kwargs"])
 
+        return_tensors = output_kwargs["common_kwargs"].get("return_tensors")
         if text is not None and videos is not None:
-            encoding["pixel_values"] = image_features.pixel_values
-            return encoding
+            return BatchFeature(data=dict(**encoding, **image_features), tensor_type=return_tensors)
         elif text is not None:
-            return encoding
+            return BatchFeature(data=dict(**encoding), tensor_type=return_tensors)
         else:
-            return BatchEncoding(data=dict(**image_features), tensor_type=return_tensors)
+            return BatchFeature(data=dict(**image_features), tensor_type=return_tensors)
```

Comment:
We can simplify this and other processors by doing only and assigning 

```python
encoding = image_features ={}
# process inputs if present here

BatchFeature(data=dict(**encoding, **image_features), tensor_type=return_tensors)
```

Reply:
@zucchini-nlp I've implemented it slightly differently, but it's now a lot cleaner than before

@yonigozlan @molbap you guys might wanna take a look too